frontend/amundsen_application/static/js/pages/TableDetailPage/WriterDropdown/index.tsx

```diff
@@ -0,0 +1,166 @@
+    (kind) => kind.toLowerCase() === PRODUCING
+  );
+  const consumingKind = appKinds.filter(
+    (kind) => kind.toLowerCase() === CONSUMING
+  );
+
+  const sortedKinds: string[] = [];
+  sortedKinds.push(...producingKind, ...consumingKind);
+  sortedKinds.push(
+    ...appKinds.filter(
+      (kind) =>
+        kind.toLowerCase() !== PRODUCING && kind.toLowerCase() !== CONSUMING
+    )
+  );
+  return sortedKinds;
+};
+
+const getMenuItem = (app: TableWriter, menuItemClassName) => {
+  if (app.name.toLowerCase() === AIRFLOW) {
+    const [dagId, ...task] = app.id.split('/');
+    const taskId = task.join('/');
+    return (
+      <MenuItem
+        className={menuItemClassName}
+        href={app.application_url}
+        onClick={logClick}
+        target="_blank"
+      >
+        <div className="writer-dropdown-menu-item-row airflow-app">
+          <span className="section-title">{DAG_LABEL}</span>
+          <span className="menu-item-content">{dagId || NONE_VALUE}</span>
+        </div>
+        <div className="writer-dropdown-menu-item-row airflow-app">
+          <span className="section-title">{TASK_LABEL}</span>
+          <span className="menu-item-content">{taskId || NONE_VALUE}</span>
+        </div>
+      </MenuItem>
+    );
+  }
+  return (
+    <MenuItem
+      className={menuItemClassName}
+      href={app.application_url}
+      onClick={logClick}
+      target="_blank"
+    >
+      <div className="writer-dropdown-menu-item-row">
+        <span className="menu-item-content">
+          {app.id ? app.id : NONE_VALUE}
+        </span>
+      </div>
+    </MenuItem>
+  );
+};
+
+const WriterDropdown: React.FC<WriterDropdownProps> = ({
+  tableApps,
+}: WriterDropdownProps) => {
+  const tableAppName = tableApps[0].name.toLowerCase();
+
+  let image = '';
+  switch (tableAppName) {
+    case AIRFLOW:
+      image = '/static/images/airflow.jpeg';
+      break;
+    case DATABRICKS:
+      image = '/static/images/icons/logo-databricks.png';
+      break;
+    default:
+      image = '/static/images/icons/Application.svg';
+  }
+
+  tableApps.sort(
+    (a, b) => a.name.localeCompare(b.name) || a.id.localeCompare(b.id)
+  );
+  const appNames: string[] = [...new Set(tableApps.map((app) => app.name))];
+
+  const menuItems: JSX.Element[] = [];
+  let appsDefinedCount = 0;
+  appNames.forEach((name, nameIdx) => {
+    const appKinds = getSortedAppKinds(
+      tableApps.filter((app) => app.name === name)
+    );
+    appKinds.forEach((kind, kindIdx) => {
+      const sectionTitle = isAirflowOrDatabricksApp(tableApps[0].name)
+        ? kind
+        : name + ' - ' + kind;
+      menuItems.push(
+        <h5 className="writer-dropdown-menu-title">{sectionTitle}</h5>
+      );
+      menuItems.push(
+        ...tableApps
+          .filter(
+            (app) =>
+              app.name === name &&
+              ((app.kind && app.kind === kind) ||
+                (!app.kind && kind === PRODUCING))
```

Comment:
Could you summarize this predicate into something human readable?
Maybe `hasSameNameAndKind` or similar?